docs/emitters/json-schema/reference/index.mdx

```diff
@@ -58,3 +58,31 @@ npm install --save-peer @typespec/json-schema
+
+The default behavior of this emitter for a model named `Widget` is to produce a schema with a `$id` of `Widget.yaml`, which allows the actual base URI of the document to differ depending on where it is retrieved from, and therefore should allow resolving relative references to other schemas whether loaded from disk or retrieved over http.
+:::
+
+When JSON Schema types reference other JSON Schema types, those references are created using a `$ref` with a relative reference (see the note above for how this works in JSON Schema).
+
+How this emitter handles data types which aren't JSON Schema types can be controlled using the `emitAllModels` and `emitAllRefs` options.
+
+By default, this emitter does not define schemas for such data types. Instead, the schemas for them are placed in the `$defs` of any JSON Schema types that reference them. The schemas do not define either `$id` or `$schema` metadata keywords. The `$defs` are referenced using a `$ref` with URI fragment containing a JSON Pointer.
+
+If you want to treat all TypeSpec types as JSON Schema types (even if they don't have the `@jsonSchema` decorator), you can set the `emitAllModels` option to true. With this set, every data type in your TypeSpec program will get its own schema file, and all references will be relative URIs.
+
+If you want to treat all TypeSpec types referenced from JSON Schema types as JSON Schema types (even if they don't have the `@jsonSchema` decorator), you can set the `emitAllRefs` option to true. With this set, data types which are referenced from JSON Schema types will get their own schema file, and all references to them will be relative URIs.
+
+## Bundling behavior
+
+By default, this emitter will produce separate schema files for each JSON Schema type in your program. However, if you prefer to bundle all of your schemas into a single file, you can set the `bundleId` option to the id you want to use for your bundle. You will now get a single file containing all of your schemas.
+
+Note that bundling does not affect any references within the bundled schemas. In particular, references between JSON Schema types still use relative URIs. As such, correctly resolving these references to the local file requires JSON Schema implementations to support bundling as defined in the JSON Schema 2020-12 specification.
+
+Note also that this does not affect the behavior of non-JSON Schema data types. In particular, by default, non-JSON Schema data types are inlined into each referencing schema's `$defs` object. The `emitAllModels` and `emitAllRefs` options can be used to turn these inlined `$defs` into `$defs` in the bundle.
```

Comment:
I went back and forth on what to call these types. I actually had your proposal but I went with what I have because often the types are not directly annotated, the decorator is on a parent namespace somewhere. But I'll take your feedback to heart and just roll with this terminology (and maybe write some prose to clarify).

Reply:
Ah, I see your point. The argument about placing `@jsonSchema` on a namespace makes sense. Maybe there should be some preface to the docs explaining what `JSON Schema data type` means (clarification of this term).

Reply:
The section starts with:

> When a TypeSpec data type has the `@jsonSchema` decorator or is declared inside a namespace with that decorator, it is considered a _JSON Schema type_.

Maybe I can beef this up somehow?

Reply:
Aha, I see, I somehow missed then. Maybe a separate header `# JSON schema type` would help

Reply:
Oh, I guess I somehow missed this. Maybe moving that whole first paragraph to a separate header `# JSON schema type` would emphasize it.

Reply:
I put it in a separate paragraph which should hopefully emphasize it enough (but feel free to propose tweaks!)